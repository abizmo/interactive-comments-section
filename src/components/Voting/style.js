import styled from 'styled-components';

export const Wrapper = styled.div`
  align-items: center;
  background: ${({ theme }) => theme.colors['neutral-200']};
  border-radius: .5rem;
  display: flex;
  gap: .5rem;
  padding: .5rem 1rem;
  width: fit-content;

  & p {
    color: ${({ theme }) => theme.colors['primary-500']};
    font-weight: 500;
    text-align: center;
    width: 4ch;
  }
`;

export const Button = styled.button`
  aspect-ratio: 1;
  background: transparent;
  border: none;
  cursor: pointer;
  display: grid;
  line-height: 1.1;
  padding-inline: .5rem;
  place-items: center;
  transition: opacity 300ms ease-in-out;

  &:hover:enabled path {
    fill: ${({ theme }) => theme.colors['primary-500']};
  }

  &:disabled {
    cursor: not-allowed;
    opacity: .5;
  }
`;
