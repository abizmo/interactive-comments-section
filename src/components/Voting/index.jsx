import React from 'react';
import PropTypes from 'prop-types';

import { ReactComponent as Minus } from '../../assets/icons/minus.svg';
import { ReactComponent as Plus } from '../../assets/icons/plus.svg';
import { Button, Wrapper } from './style';

function Voting({ onVote, votes }) {
  return (
    <Wrapper data-testid="voting">
      <Button onClick={() => onVote((prv) => prv + 1)} type="button">
        <Plus />
        <span className="sr-only">Plus</span>
      </Button>
      <p>{votes}</p>
      <Button disabled={votes === 0} onClick={() => onVote((prv) => prv - 1)} type="button">
        <Minus />
        <span className="sr-only">Minus</span>
      </Button>
    </Wrapper>
  );
}

Voting.propTypes = {
  onVote: PropTypes.func.isRequired,
  votes: PropTypes.number.isRequired,
};

export default Voting;
