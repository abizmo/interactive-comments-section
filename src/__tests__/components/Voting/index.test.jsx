import React from 'react';
// eslint-disable-next-line no-unused-vars
import { fireEvent, render, screen } from '@testing-library/react';

import Voting from '../../../components/Voting';
import Theme from '../../../theme';

test('renders and MatchSnapshot', () => {
  const handleVote = jest.fn();

  const { container } = render(
    <Theme>
      <Voting onVote={handleVote} votes={5} />
    </Theme>,
  );

  expect(screen.getByText('0005')).toBeTruthy();
  expect(screen.getByRole('widget')).toHaveStyle('background: hsl( 228, 33%, 97%)');

  fireEvent.click(screen.getByText('Plus'));
  expect(handleVote).toHaveBeenCalled();

  fireEvent.click(screen.getByText('Minus'));
  expect(handleVote).toHaveBeenCalled();

  expect(container).toMatchSnapshot();
});
