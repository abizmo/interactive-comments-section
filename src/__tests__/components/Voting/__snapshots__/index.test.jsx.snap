// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and MatchSnapshot 1`] = `
<div>
  <div
    class="sc-bczRLJ jHukZu"
    role="widget"
  >
    <button
      class="sc-gsnTZi czLDEY"
      type="button"
    >
      <svg>
        plus.svg
      </svg>
      <span
        class="sr-only"
      >
        Plus
      </span>
    </button>
    <p>
      5
    </p>
    <button
      class="sc-gsnTZi czLDEY"
      type="button"
    >
      <svg>
        minus.svg
      </svg>
      <span
        class="sr-only"
      >
        Minus
      </span>
    </button>
  </div>
</div>
`;
